### PR TITLE
ironware_config module

### PR DESCRIPTION
**Pull request by paulquack**
##### SUMMARY

Configuration management for Brocade Ironware routers

##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME

ironware_config

##### ANSIBLE VERSION

```
ansible 2.5.0 (ironware_facts 83955157c0) last updated 2017/10/26 22:18:00 (GMT +1100)
  config file = /Users/paul.baker/.ansible.cfg
  configured module search path = [u'/Users/paul.baker/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/paul.baker/ansible/lib/ansible
  executable location = /Users/paul.baker/ansible/bin/ansible
  python version = 2.7.13 (default, Dec 18 2016, 07:03:39) [GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.42.1)]
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

**Comment by paulquack** [comment]
I've included a note along with my examples for this module which references a feature request I've submitted. Please let me know if this should be removed from documentation section or whether I can resolve it by submitting a PR to extend NetworkConfig object as described in [31568](https://github.com/ansible/ansible/issues/31568)

```
# Note: Network modules do not support loading hirachial config commands
# matching by line, the following example would currently fail if the C(match)
# parameter is set to I(line), unfortunately in some instances it may still fail
# even when C(match) is set to I(block) if either C(parents) or C(lines) include
# parameters which can only be specified when creating a new VLL. Hopefully this
# can be handled better in some future version. See issue #31568 for more detail.
- ironware_config:
    host: "{{ inventory_hostname }}"
    lines:
      - to 10.0.0.1
      - adaptive
      - reoptimize-timer 1800
      - frr
      -  facility-backup
      - revert-timer 300
      - enable
    match: block
    parents: ['router mpls', 'lsp test-lsp']
    provider: "{{ cli }}"
```

File: lib/ansible/modules/network/ironware/ironware_config.py [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+#       transport and authentication to the node.
+---
+vars:
+  cli:
+    host: "{{ inventory_hostname }}"
+    username: username
+    password: secret
+    authorize: yes
+    auth_pass: supersecret
+    transport: cli
```
Comment (ganeshrn) [inline lib/ansible/modules/network/ironware/ironware_config.py]: `transport` is not part of ironware provider [spec](https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/ironware.py#L39). Is this option required?

Comment (paulquack) [reply lib/ansible/modules/network/ironware/ironware_config.py]: Looks like `transport` is referenced in documentation of several networking modules but only used by a handful. E.g. some asa and ios modules reference it. 

It also seems the trend is not to talk about `provider` at all in EXAMPLES, instead purely relying on the documentation derived from the module argument spec. So do you agree that I remove this note and references to `provider` from EXAMPLES?

**Review by ganeshrn — APPROVED** [review]
Looks good.

**Comment by ganeshrn** [comment]
I think the best approach here would be to remove the example (and the comment) as it will fail with current code. 

You can raise a feature request PR to fix the issue #31568,  after the PR is merged you can re-add the example.

**Comment by paulquack** [comment]
As suggested I have removed the bad example.

**Comment by ganeshrn** [comment]
@paulquack Thank you!

**Comment by paulquack** [comment]
shipit

**Comment by paulquack** [comment]
bot_status

**Review by gundalow — APPROVED** [review]

**Comment by gundalow** [comment]
Thank you for the module and tests. 
Merged into `devel`, will be released in  Ansible 2.5.0.

**Pull request merged**